e2e/cypress/integration/messaging/mention_autocomplete_overlap_spec.js

```diff
@@ -0,0 +1,50 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+
+// ***************************************************************
+// - [number] indicates a test step (e.g. 1. Go to a page)
+// - [*] indicates an assertion (e.g. * Check the title)
+// - Use element ID when selecting an element. Create one if none.
+// ***************************************************************
+
+import * as TIMEOUTS from '../../fixtures/timeouts';
+
+describe('Messaging', () => {
+    before(() => {
+        // # Login and go to /
+        cy.apiLogin('user-1');
+        cy.visit('/');
+    });
+
+    it('M18667-At-mention user autocomplete does not overlap with channel header when drafting a long message containing a file attachment', () => {
+        cy.visit('/ad-1/channels/town-square');
+
+        // # Upload file
+        cy.fileUpload('#fileUploadInput');
+
+        // # Create and then type message to use
+        let message = 'h{shift}';
+        cy.get('#post_textbox').clear();
+        for (let i = 0; i < 12; i++) {
+            message += '{enter}h';
+        }
+        cy.get('#post_textbox').type(message);
+
+        // # Put cursor on top and add the mention
+        let returnMessage = '';
+        for (let i = 0; i < 12; i++) {
+            returnMessage += '{uparrow}';
+        }
+        cy.get('#post_textbox').type(returnMessage + '{leftarrow}@');
+
+        cy.get('#suggestionList').then((list) => {
+            cy.get('#channel-header').then((header) => {
+                // # Wait for suggestions to be fully loaded
+                cy.wait(TIMEOUTS.TINY).then(() => {
+                    // * Check overlap
+                    expect(header[0].getBoundingClientRect().bottom).be.lte(list[0].getBoundingClientRect().top);
```

Comment:
It took me a while to understand this assertion. I would suggest the following assertion instead to have it be a little more clear:
```suggestion
                    expect(list[0].getBoundingClientRect().top >= header[0].getBoundingClientRect().bottom).be.true;
```
What do you think?

Reply:
My option is a bit more verbose on Cypress. With your suggestion, you will only see on the action list something like "expect true to be true" or "expect false to be true". With my option you will see "expect x to be at most y". Still not the most useful information, but at least a bit more verbose. Either way, if you find it not needed, I can change it without any problem.

Reply:
> My option is a bit more verbose on Cypress. With your suggestion, you will only see on the action list something like "expect true to be true" or "expect false to be true". With my option you will see "expect x to be at most y". Still not the most useful information, but at least a bit more verbose. Either way, if you find it not needed, I can change it without any problem.

Ah okay. I understand what you are saying. In that case, then maybe we can change it to
```suggestion
                    expect(list[0].getBoundingClientRect().top).to.be.at.least(header[0].getBoundingClientRect().bottom);
```
I would personally avoid the shorthands such as `lte` as they may not be that clear. I prefer using `at.least` since the `list` rectangle is the one moving around and the `header` rectangle is stationary so it makes more sense to have that as the first parameter.